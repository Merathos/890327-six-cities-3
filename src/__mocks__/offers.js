export default [
  {
    id: `41324`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3909553943508, 4.85309666406198],
    zoom: 10,
    title: `Beautiful & luxurious apartment at great location`,
    previewImg: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 3,
    isBookmarked: true,
    isPremium: false,
    price: 120,
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    hostStatus: true,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `73645`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.369553943508, 4.85309666406198],
    title: `Wood and stone place`,
    previewImg: `img/apartment-02.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 2,
    maxAdults: 3,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    type: `Private room`,
    rating: 4,
    isBookmarked: false,
    isPremium: true,
    price: 80,
    hostName: `Max`,
    hostAvatar: `img/avatar-max.jpg`,
    hostStatus: false,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `23455`,
    city: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3909553943508, 4.929309666406198],
    title: `Canal View Prinsengracht`,
    previewImg: `img/apartment-03.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 3,
    maxAdults: 4,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 5,
    isBookmarked: true,
    isPremium: false,
    price: 132,
    hostName: `Angelina`,
    hostAvatar: `img/avatar-angelina.jpg`,
    hostStatus: false,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  But where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `84578`,
    city: {
      name: `Paris`,
      coords: [53.370216, 4.895168],
      zoom: 10
    },
    coords: [52.3809553943508, 4.939309666406198],
    title: `Nice, cozy, warm big bed apartment`,
    previewImg: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    bedroomsAmount: 4,
    maxAdults: 5,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Private room`,
    rating: 2,
    isBookmarked: false,
    isPremium: true,
    price: 180,
    hostName: `Max`,
    hostAvatar: `img/avatar-max.jpg`,
    hostStatus: true,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera.`
  }
];
