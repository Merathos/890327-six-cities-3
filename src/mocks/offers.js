export default [
  {
    id: `41324`,
    name: `Beautiful &amp; luxurious apartment at great location`,
    picture: `apartment-01`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 1,
    maxAdults: 2,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 3,
    isBookmarked: true,
    isPremium: false,
    price: 120,
    hostName: `Angelina`,
    hostAvatar: `avatar-angelina`,
    hostStatus: `pro`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `73645`,
    name: `Wood and stone place`,
    picture: `apartment-02`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 2,
    maxAdults: 3,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`
    ],
    type: `Private room`,
    rating: 4,
    isBookmarked: false,
    isPremium: true,
    price: 80,
    hostName: `Max`,
    hostAvatar: `avatar-max`,
    hostStatus: ``,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  An independent House, strategically located between Rembrand Square and National Opera,
                  but where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `23455`,
    name: `Canal View Prinsengracht`,
    picture: `apartment-03`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 3,
    maxAdults: 4,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Fridge`
    ],
    type: `Apartment`,
    rating: 5,
    isBookmarked: true,
    isPremium: false,
    price: 132,
    hostName: `Angelina`,
    hostAvatar: `avatar-angelina`,
    hostStatus: ``,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  But where the bustle of the city comes to rest in this alley flowery and colorful.`
  },
  {
    id: `84578`,
    name: `Nice, cozy, warm big bed apartment`,
    picture: `apartment-01`,
    photos: [`room`, `room`, `room`, `room`, `room`, `room`],
    bedroomsAmount: 4,
    maxAdults: 5,
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    type: `Private room`,
    rating: 2,
    isBookmarked: false,
    isPremium: true,
    price: 180,
    hostName: `Max`,
    hostAvatar: `avatar-max`,
    hostStatus: `pro`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                  The building is green and from 18th century.
                  An independent House, strategically located between Rembrand Square and National Opera.`
  }
];
