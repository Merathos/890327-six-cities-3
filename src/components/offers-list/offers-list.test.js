import React from "react";
import renderer from "react-test-renderer";
import OffersList from "./offers-list.jsx";
import offers from "../../__mocks__/offers";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore([]);

const cities = [
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
  {
    name: `Paris`,
    coords: [4134, 123123],
    zoom: 10
  },
];

const hoveredCard = {
  coords: [123123, 123123]
};

it(`Render Offers list`, () => {
  const store = mockStore({
    cities,
    currentCity: {
      name: `Amsterdam`,
      coords: [52.370216, 4.895168],
      zoom: 10
    },
    allOffers: offers,
    offersByCity: offers.filter((offer) => offer.city.name === `Amsterdam`),
    offersByCitySorted: offers.filter((offer) => offer.city.name === `Amsterdam`),
    currentSortType: `Popular`,
    hoveredCard
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <OffersList
            offers={offers}
            handleRentHeaderClick={() => {}}
          />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
