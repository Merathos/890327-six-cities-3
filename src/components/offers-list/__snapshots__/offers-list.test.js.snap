// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Offers list 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            523
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          Ea aute voluptate amet magna id qui
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/img/apartment-02.jpg.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            23
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          ex occaecat consectetur enim velit eu.
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/img/apartment-03.jpg.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            857
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button "
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          Cillum Lorem cillum consectetur
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/img/apartment-01.jpg.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            188
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button place-card__bookmark-button--active"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="#"
        >
          quis do anim aliqua et eiusmod sit aute officia.
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>
</div>
`;
